pkg/controllermanager/controller/shoot/shoot_control.go

```diff
@@ -143,46 +113,33 @@ func (c *Controller) reconcileShootKey(key string) error {
 	// Ignore Shoots which do not have the gardener finalizer.
 	if shoot.DeletionTimestamp != nil && !sets.NewString(shoot.Finalizers...).Has(gardenv1beta1.GardenerName) {
 		shootLogger.Debug("Do not need to do anything as the Shoot does not have my finalizer")
-		c.scheduler.Delete(shootID)
 		c.getShootQueue(shoot).Forget(key)
 		return nil
 	}
 
-	shootElement, err := c.newShootElement(shoot)
-	if err != nil {
-		return err
+	seedName := shoot.Spec.Cloud.Seed
+	if seedName == nil {
+		return fmt.Errorf("shoot has not yet been scheduled on a seed")
```

Comment:
Previously we were recording an  event on the shoot when the seed is missing to indicate that it is waiting to be scheduled. Now we return an error - is that the wanted behaviour?

Reply:
Returning an error here will result in requeueing and a retry with exponential backoff, however, the shoot object won't get an error status. I think this is exactly what we want. However, we should also still send the event that we don't reconcile because the shoot is not scheduled to a seed. OK, @adracus ?

Reply:
@danielfoehrKn you're right - We changed behavior here. I reverted it back to the original state (no exponential backup on unschedulable but 10 second retry) and also adapted error message and event.